editors/sc-ide/widgets/code_editor/editor.hpp

```diff
@@ -41,6 +41,16 @@ class GenericCodeEditor : public QPlainTextEdit {
 
 public:
     GenericCodeEditor(Document*, QWidget* parent = NULL);
+    ~GenericCodeEditor();
+
+    // Rule of 5 -- with an explicit dtor, make sure that copy ctor, move ctor,
+    // copy assign, and move assign are deleted.
+    // If these must be implemented in the future, make sure they take care of
+    // mDoc->lastActiveEditor() appropriately. See source of dtor method.
+    GenericCodeEditor(const GenericCodeEditor& other) = delete;
+    GenericCodeEditor(GenericCodeEditor&& other) = delete;
+    GenericCodeEditor& operator=(const GenericCodeEditor& other) = delete;
+    GenericCodeEditor& operator=(GenericCodeEditor&& other) = delete;
```

Comment:
note for the future -- typically these are ordered as copy ctor, copy assign, move ctor, move assign. that's because they're all viewed of sort-of constructors, and grouping things by copy vs move semantics is easier to read logically.